specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-03-01-preview/cosmos-db.json

```diff
@@ -0,0 +1,8956 @@
+          "$ref": "#/definitions/SqlContainerResource"
+        },
+        "options": {
+          "description": "A key-value pair of options to be applied for the request. This corresponds to the headers sent with the request.",
+          "$ref": "#/definitions/CreateUpdateOptions"
+        }
+      },
+      "required": [
+        "resource"
+      ]
+    },
+    "SqlStoredProcedureCreateUpdateParameters": {
+      "description": "Parameters to create and update Cosmos DB storedProcedure.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties to create and update Azure Cosmos DB storedProcedure.",
+          "$ref": "#/definitions/SqlStoredProcedureCreateUpdateProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ARMResourceProperties"
+        }
+      ],
+      "required": [
+        "properties"
+      ]
+    },
+    "SqlStoredProcedureCreateUpdateProperties": {
+      "description": "Properties to create and update Azure Cosmos DB storedProcedure.",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The standard JSON format of a storedProcedure",
+          "$ref": "#/definitions/SqlStoredProcedureResource"
+        },
+        "options": {
+          "description": "A key-value pair of options to be applied for the request. This corresponds to the headers sent with the request.",
+          "$ref": "#/definitions/CreateUpdateOptions"
+        }
+      },
+      "required": [
+        "resource"
+      ]
+    },
+    "SqlUserDefinedFunctionCreateUpdateParameters": {
+      "description": "Parameters to create and update Cosmos DB userDefinedFunction.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties to create and update Azure Cosmos DB userDefinedFunction.",
+          "$ref": "#/definitions/SqlUserDefinedFunctionCreateUpdateProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ARMResourceProperties"
+        }
+      ],
+      "required": [
+        "properties"
+      ]
+    },
+    "SqlUserDefinedFunctionCreateUpdateProperties": {
+      "description": "Properties to create and update Azure Cosmos DB userDefinedFunction.",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The standard JSON format of a userDefinedFunction",
+          "$ref": "#/definitions/SqlUserDefinedFunctionResource"
+        },
+        "options": {
+          "description": "A key-value pair of options to be applied for the request. This corresponds to the headers sent with the request.",
+          "$ref": "#/definitions/CreateUpdateOptions"
+        }
+      },
+      "required": [
+        "resource",
+        "options"
```

Comment:
This is a breaking change from the previous stable api-version

Reply:
Removed options from required list.

Reply:
Hmm, I had reverted this. I wonder if I missed this one line, but it appears in >10 places, where I fixed it. Thanks for catching this. 
Fred: can you please check if this is one place where I missed it or it happens throughout. If latter, then maybe my commits were over-written, if former, can you please remove options from required.

Reply:
Checked. It was just the one place.